src/linagora.esn.calendar/app/calendar-configuration/calendar-configuration-tab-main/calendar-configuration-tab-main.controller.js

```diff
@@ -1,119 +1,138 @@
+
+  ///////////
+  function $onInit() {
+    self.publicRights = [
+      {
+        value: CAL_CALENDAR_PUBLIC_RIGHT.READ_WRITE,
+        name: CAL_CALENDAR_PUBLIC_RIGHT.READ_WRITE_LABEL_LONG
+      },
+      {
+        value: CAL_CALENDAR_PUBLIC_RIGHT.READ,
+        name: CAL_CALENDAR_PUBLIC_RIGHT.READ_LABEL_LONG
+      },
+      {
+        value: CAL_CALENDAR_PUBLIC_RIGHT.PRIVATE,
+        name: CAL_CALENDAR_PUBLIC_RIGHT.PRIVATE_LABEL_LONG
+      }
+    ];
+
+    !self.newCalendar && performExternalCalendarOperations(isExternalCalendar());
+    self.canModifyPublicSelection = _canModifyPublicSelection();
+    self.canExportIcs = canExportIcs();
+    self.canDeleteCalendar = canDeleteCalendar();
+    self.isResource = self.calendar.type === CAL_CALENDAR_TYPE.RESOURCE;
+
+    if (!self.newCalendar && self.calendar) {
+      // Used in the template to show the calendar DAV URL.
+      calCalDAVURLService.getCalendarURL(self.calendar).then(url => {
+        self.caldavurl = url;
+      });
 
-      !self.newCalendar && performExternalCalendarOperations(isExternalCalendar());
-      self.canModifyPublicSelection = _canModifyPublicSelection();
-      self.canExportIcs = canExportIcs();
-      self.isResource = self.calendar.type === CAL_CALENDAR_TYPE.RESOURCE;
+      self.calendarToExport = self.calendar.isSubscription() ? self.calendar.source : self.calendar;
+    }
+  }
 
-      if (!self.newCalendar && self.calendar) {
-        // Used in the template to show the calendar DAV URL.
-        calCalDAVURLService.getCalendarURL(self.calendar).then(url => {
-          self.caldavurl = url;
-        });
+  function isExternalCalendar() {
+    return self.calendar.isShared(session.user._id) || self.calendar.isSubscription();
+  }
 
-        self.calendarToExport = self.calendar.isSubscription() ? self.calendar.source : self.calendar;
-      }
-    }
+  function openDeleteConfirmationDialog() {
+    calCalendarDeleteConfirmationModalService(self.calendar, removeCalendar);
+  }
 
-    function isExternalCalendar() {
-      return self.calendar.isShared(session.user._id) || self.calendar.isSubscription();
-    }
+  function openGetSecretLinkConfirmationDialog() {
+    calCalendarSecretAddressConfirmationModalService(self.calendar, createSecretLinkWithToken);
+  }
 
-    function unsubscribe() {
-      calendarService.unsubscribe(self.calendarHomeId, self.calendar).then(function() {
-        $state.go('calendar.main');
-      });
-    }
+  function unsubscribe() {
+    calendarService.unsubscribe(self.calendarHomeId, self.calendar).then(function() {
+      $state.go('calendar.main');
+    });
+  }
 
-    function canExportIcs() {
-      return !self.newCalendar && calUIAuthorizationService.canExportCalendarIcs(self.calendar, session.user._id);
-    }
+  function removeCalendar() {
+    calendarService.removeCalendar(self.calendarHomeId, self.calendar).then(function() {
+      $state.go('calendar.main');
+    });
+  }
 
```

Comment:
I smell something fishy here. Why is this function added? We don't do anything about the calendar removal functionality right?